3-enrich/stream-enrich/core/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/utils.scala

```diff
@@ -135,33 +137,31 @@ object utils {
     provider: AWSCredentialsProvider,
     uri: URI,
     targetFile: File
-  ): Either[Throwable, Int] = {
-    val s3Client = AmazonS3ClientBuilder.standard().withCredentials(provider).build()
-    val bucketName = uri.getHost
-    val key = uri.getPath match { // Need to remove leading '/'
-      case s if s.length > 0 && s.charAt(0) == '/' => s.substring(1)
-      case s => s
-    }
+  ): Either[Throwable, Int] =
     Either.catchNonFatal {
+      val s3Client = AmazonS3ClientBuilder.standard().withCredentials(provider).build()
+      val bucketName = uri.getHost
+      val key = uri.getPath match { // Need to remove leading '/'
+        case s if s.length > 0 && s.charAt(0) == '/' => s.substring(1)
+        case s => s
+      }
       s3Client.getObject(new GetObjectRequest(bucketName, key), targetFile)
       0
     }
```

Comment:
We could have more fine-grained error control with a `for {} yield` and individual `Either` with `.leftMap` rather than a global `Either.catchNonFatal {` with everything inside and return `Either[String, Int]` with a clear error message rather than a `Throwable`.